sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -1272,6 +1273,56 @@ class SQLContext(@transient val sparkContext: SparkContext)
     }
   }
 
+  SQLContext.setLastInstantiatedContext(self)
```

Comment:
@rxin  Design question. Should contexts created directly (that is, not through `getOrCreate`) also be automatically considered for the singleton?